NitroxClient/MonoBehaviours/DiscordRP/DiscordRPController.cs

```diff
@@ -0,0 +1,153 @@
+        public static DiscordRPController Main
+        {
+            get
+            {
+                if (main == null)
+                {
+                    main = new GameObject("DiscordController").AddComponent<DiscordRPController>();
+                }
+                return main;
+            }
+        }
+
+        public void ReadyCallback(ref DiscordRpc.DiscordUser connectedUser) => Log.Info("[Discord] Ready");
+
+        public void DisconnectedCallback(int errorCode, string message) => Log.Info($"[Discord] Disconnected: {errorCode} <=> {message}");
+
+        public void ErrorCallback(int errorCode, string message) => Log.Error($"[Discord] Error: {errorCode} <=> {message}");
+
+        public void SpectateCallback(string secret) => Log.Info($"[Discord] Spectate: {secret}");
+
+        public void JoinCallback(string secret)
+        {
+            Log.Info($"[Discord] Join: {secret}");
+            if (SceneManager.GetActiveScene().name == "StartScreen")
+            {
+                MainMenuMultiplayerPanel.Main.OpenJoinServerMenu(secret);
+            }
+            else
+            {
+                Log.InGame("Please enter the main-menu if you want to join a session.");
+                Log.Warn("[Discord] Warn: Can't join a server outside of the main-menu.");
+            }
+        }
+
+        public void RequestCallback(ref DiscordRpc.DiscordUser request)
+        {
+            if (!ShowingWindow)
+            {
+                Log.Info($"[Discord] JoinRequest: Name:{request.username}#{request.discriminator} UserID:{request.userId}");
+                DiscordJoinRequestGui acceptRequest = gameObject.AddComponent<DiscordJoinRequestGui>();
+                acceptRequest.Request = request;
+                ShowingWindow = true;
+            }
+            else
+            {
+                Log.Debug("[Discord] Request window is already active.");
+            }
+        }
+
+        private void Awake()
+        {
+            DontDestroyOnLoad(gameObject);
+        }
+
+        private void Update()
+        {
+            DiscordRpc.RunCallbacks();
+        }
+
+        private void OnEnable()
+        {
+            Log.Info("[Discord] Init");
+            handlers = new DiscordRpc.EventHandlers
+            {
+                readyCallback = ReadyCallback
+            };
+            handlers.disconnectedCallback += DisconnectedCallback;
+            handlers.errorCallback += ErrorCallback;
+            handlers.joinCallback += JoinCallback;
+            handlers.spectateCallback += SpectateCallback;
+            handlers.requestCallback += RequestCallback;
+            DiscordRpc.Initialize(APPLICATION_ID, ref handlers, true, "");
+        }
+
+        private void OnDisable()
+        {
+            Log.Info("[Discord] Shutdown");
+            DiscordRpc.Shutdown();
+        }
+
+        public static void InitDiscordRPInGame(string username, int playerCount, string ipAddressPort)
```

Comment:
maybe just `InitializeInGame` as DiscordRP is implied by the class.  (Same with all of the other methods)